paddle/fluid/operators/detail/rpc_processor.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "grpc++/grpc++.h"
+#include "grpc++/support/byte_buffer.h"
+#include "grpc/support/log.h"
+#include "paddle/fluid/framework/blocking_queue.h"
+#include "paddle/fluid/framework/data_type.h"
+#include "paddle/fluid/framework/lod_tensor.h"
+#include "paddle/fluid/framework/scope.h"
+#include "paddle/fluid/framework/selected_rows.h"
+#include "paddle/fluid/operators/detail/rpc_processor.h"
+#include "paddle/fluid/operators/detail/sendrecvop_utils.h"
+
+namespace paddle {
+namespace operators {
+namespace detail {
+
+void GRPCProcessorCtx::SetExit() {
+  LOG(WARNING) << "GRPCProcessorCtx SetExit!";
+  std::unique_lock<std::mutex> lock(mutex_);
+  exit_flag_ = true;
+  condition_send_.notify_all();
+  condition_get_.notify_all();
+}
+
+void GRPCProcessorCtx::IncreaseBatchBarrierSend() {
+  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
A `condition variable` should used with a mutex as a pair, and `mutex_` is used anywhere, maybe you can use multiple mutex to project these `condition variable`.

Reply:
Reserve only one! Thanks! Done!